tailnet/conn.go

```
@@ -439,6 +443,18 @@ func (c *Conn) SetAddresses(ips []netip.Prefix) error {
 	return nil
 }
 
+func (c *Conn) AddDNSHosts(hosts map[dnsname.FQDN][]netip.Addr) {
+	if c.dnsConfigurator == nil {
+		c.logger.Warn(context.Background(), "no DNSConfigurator set, rejecting custom DNS hosts")
```

Let's return an error, rather than just log a warning.